src/Microsoft.ML.Transforms/GroupTransform.cs

```diff
@@ -180,35 +180,42 @@ public override RowCursor[] GetRowCursorSet(Func<int, bool> predicate, int n, Ra
         }
 
         /// <summary>
+        /// The output schema collects columns used to group columns (aka group-key columns) columns being grouped from the source scheam.
+        /// In an output schema, group-key columns are followed by grouped columns. For example, if column "Age" is used to group "UserId" column,
+        /// the first column and the second column in the output schema produced by <see cref="GroupTransform"/> would be "Age" and "UserId," respectively.
         /// For group columns, the schema information is intact.
         ///
         /// For keep columns, the type is Vector of original type and variable length.
         /// The only metadata preserved is the KeyNames and IsNormalized.
         ///
         /// All other columns are dropped.
         /// </summary>
-        private sealed class GroupSchema : ISchema
+        private sealed class GroupBinding
         {
             private static readonly string[] _preservedMetadata =
                 new[] { MetadataUtils.Kinds.IsNormalized, MetadataUtils.Kinds.KeyValues };
```

Comment:
either use it in BuildOutputSchema or delete it.

Reply:
Deleted.

---
In reply to: [245810211](https://github.com/dotnet/machinelearning/pull/2060#discussion_r245810211) [](ancestors = 245810211)